### ISSUE TITLE
PL 1.4.0 broke `training_epoch_end`

### ISSUE THREAD

**Issue by tmlabonte** [PyTorchLightning/pytorch-lightning#8660]
## 🐛 Bug

In PL 1.3.0, `training_epoch_end` would accumulate the training losses properly, e.g.

`[{'loss': tensor(30.8064, device='cuda:0')}, {'loss': tensor(41.2687, device='cuda:0')}, {'loss': tensor(85.0726, device='cuda:0')}, {'loss': tensor(81.9442, device='cuda:0')}, {'loss': tensor(85.0309, device='cuda:0')}]`.

However, in PL 1.4.0, `training_epoch_end` overwrites all losses with the final step's loss, e.g.

`[{'loss': tensor(85.0309, device='cuda:0')}, {'loss': tensor(85.0309, device='cuda:0')}, {'loss': tensor(85.0309, device='cuda:0')}, {'loss': tensor(85.0309, device='cuda:0')}, {'loss': tensor(85.0309, device='cuda:0')}]`.

This only happens on `training_epoch_end` and not validation.

### To Reproduce

Reproduced on BoringModel: https://colab.research.google.com/drive/10-497crHO9gKdA5LSYiERIiE_-7RSnId?usp=sharing

### Expected behavior

Losses should accumulate properly.

### Environment

The environment script on BoringModel 404's (https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py), but I just used the BoringModel exactly as is.

**Comment by ananthsub** [comment]
https://github.com/PyTorchLightning/pytorch-lightning/issues/8603 - the fix has landed in the main branch and will be available in the 1.4.1 patch release

**Comment by tmlabonte** [comment]
Ah, I didn't search in the closed issues. Thanks.
